packages/peregrine/lib/talons/AccountInformationPage/useAccountInformationPage.js

```diff
@@ -0,0 +1,112 @@
+import { useCallback, useMemo, useState } from 'react';
+import { useMutation, useQuery } from '@apollo/client';
+import { useUserContext } from '../../context/user';
+
+export const useAccountInformationPage = props => {
+    const {
+        mutations: {
+            setCustomerInformationMutation,
+            changeCustomerPasswordMutation
+        },
+        queries: { getCustomerInformationQuery }
+    } = props;
+
+    const [{ isSignedIn }] = useUserContext();
+    const [shouldShowNewPassword, setShouldShowNewPassword] = useState(false);
+
+    const [isUpdateMode, setIsUpdateMode] = useState(false);
+
+    const { data: accountInformationData, error: loadDataError } = useQuery(
+        getCustomerInformationQuery,
+        {
+            skip: !isSignedIn,
+            fetchPolicy: 'cache-and-network',
+            nextFetchPolicy: 'cache-first'
+        }
+    );
+
+    const [
+        setCustomerInformation,
+        {
+            data: customerInformationUpdateData,
+            error: customerInformationUpdateError,
+            loading: isUpdatingCustomerInformation
+        }
+    ] = useMutation(setCustomerInformationMutation);
+
+    const [
+        changeCustomerPassword,
+        {
+            error: customerPasswordChangeError,
+            loading: isChangingCustomerPassword
+        }
+    ] = useMutation(changeCustomerPasswordMutation);
+
+    const initialValues = useMemo(() => {
+        if (customerInformationUpdateData) {
+            return {
+                customer: customerInformationUpdateData.updateCustomer.customer
+            };
+        }
+        if (accountInformationData) {
+            return { customer: accountInformationData.customer };
+        }
+    }, [customerInformationUpdateData, accountInformationData]);
+
+    const handleChangePassword = useCallback(() => {
+        setShouldShowNewPassword(true);
+    }, [setShouldShowNewPassword]);
+
+    const cancelUpdateMode = useCallback(() => {
+        setIsUpdateMode(false);
+        setShouldShowNewPassword(false);
+    }, [setIsUpdateMode]);
+
+    const showUpdateMode = useCallback(() => {
+        setIsUpdateMode(true);
+    }, [setIsUpdateMode]);
+
+    const handleSubmit = useCallback(
+        async formValues => {
+            try {
+                await setCustomerInformation({
```

Comment:
We should make sure this data changed before submitting this mutation. Also consider filtering out password fields.

Reply:
> We should make sure this data changed before submitting this mutation.

Good call - we have access to the customer's data in `AccountInformationPageFragment`.

>  Also consider filtering out password fields.

I commented elsewhere but you need password for email changes. We can (and should) filter out `newPassword` though.

Reply:
I have just update condition to update new customer information. Can you check help me?